pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,670 @@
+from math import log
+import itertools
+import numpy as np
+from operator import sub
+from operator import add
+import queue
+import math
+
+
+class MPLPAlg:
+    """
+    Class for performing inference using Max-Product Linear Programming Relaxations.
+    """
+
+    def __init__(self, all_cardinalities, all_factors, all_lambdas):
```

Comment:
Furthermore, this shouldn't be the way to instantiate an inference class object. Please have a look at how the exact inference classes are instantiated and modify it accordingly. **DON"T** try to implement the algorithm present in the paper as it is.